packages/core/src/payment/strategies/braintree/braintree-venmo-payment-strategy.ts

```diff
@@ -0,0 +1,156 @@
+
+            return await this._store.dispatch(
+                this._paymentActionCreator.submitPayment(paymentData),
+            );
+        } catch (error) {
+            this._handleError(error);
+        }
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    async deinitialize(): Promise<InternalCheckoutSelectors> {
+        await this._braintreePaymentProcessor.deinitialize();
+
+        return this._store.getState();
+    }
+
+    private _handleError(error: BraintreeError | Error): never {
+        if (!isBraintreeError(error)) {
+            throw error;
+        }
+
+        if (error.code === 'PAYPAL_POPUP_CLOSED') {
+            throw new PaymentMethodCancelledError(error.message);
+        }
+
+        throw new PaymentMethodFailedError(error.message);
+    }
+
+    private async _initializeBraintreeVenmo(clientToken?: string): Promise<void> {
+        if (!clientToken) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        try {
+            this._braintreePaymentProcessor.initialize(clientToken);
+            this._braintreeVenmoCheckout = await this._braintreePaymentProcessor.getVenmoCheckout();
+        } catch (error) {
+            this._handleError(error);
+        }
+    }
+
+    private async _preparePaymentData(payment: OrderPaymentRequestBody): Promise<Payment> {
+        const { nonce } = this._store
+            .getState()
+            .paymentMethods.getPaymentMethodOrThrow(payment.methodId);
+
+        if (nonce) {
+            return { ...payment, paymentData: this._formattedPayload(nonce) };
+        }
+
+        const tokenizeResult = await this._braintreeVenmoTokenize();
+        const sessionId = await this._braintreePaymentProcessor.getSessionId();
+
+        return {
+            ...payment,
+            paymentData: this._formattedPayload(
+                tokenizeResult.nonce,
+                tokenizeResult.details.email,
+                sessionId,
+            ),
+        };
+    }
+
+    private _formattedPayload(
+        token: string,
+        email?: string,
+        sessionId?: string,
+    ): FormattedPayload<PaypalInstrument> {
+        return {
+            formattedPayload: {
+                vault_payment_instrument: null,
+                set_as_default_stored_instrument: null,
+                device_info: sessionId || null,
+                paypal_account: {
+                    token,
+                    email: email || null,
+                },
+            },
+        };
+    }
+
+    private async _braintreeVenmoTokenize(): Promise<BraintreeTokenizePayload> {
+        return new Promise((resolve, reject) => {
+            this._braintreeVenmoCheckout?.tokenize(
+                async (error: BraintreeError, payload: BraintreeTokenizePayload) => {
```

Comment:
Strange that we need to do this, usually async should have await accompanied.

Since you are resolving a promise I think should be able to remove the `async` here

Reply:
`async` removed